app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -11,6 +11,8 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
       default_hostname: '',
       amqp_hostname: '',
       hawkular_hostname: '',
+      provider_options_values: {},
```

Comment:
Ah, got it :).

Well, it will probably work just fine as it is, I'm a bit reluctant to try to force you to make this super-clean when that whole screen is super-dirty already :).

That said, yes I would work that other approach better. If the structure is relatively flat, you could also `$watch` it for changes and update the model value on any change.